### PR TITLE
Warn user about unsaved changes

### LINKED ISSUE #830: As a xoder I want to be notified about unsaved edits before closing IDE so that I’d not lose my work accidentally
### Rationale

Make some big portion of work and hit IDE close. Ooops! :sob: 

Yes, we have #485. However, it’s not yet implemented and even when it would we should have a notion of some explicit milestone points that would act as current “Save Project” from a user point of view.

The feature should work in both, browser and desktop versions.

### Acceptance criteria

- [ ] If I open a project, make no changes, and try to close an IDE no questions are asked, it closes
- [ ] If I open a project, make some non-destructive changes (select/deselect items, open/close tabs), and try to close an IDE no questions are asked, it closes
- [ ] If I open a project, make changes, and try to close the IDE window, I’m asked whether I want to lose all changes
- [ ] If I open a project, make changes, and try to close the project (open another one or switch workspace), I’m asked whether I want to lose all changes

### How to implement

Use standard [onbeforeupload](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload) for browser. Use [on('close', ...)](https://github.com/electron/electron/issues/2301) for Electron.

To keep things simple use the same kind of dialogs (native) for asking on project switches.

### PR DESCRIPTION
**Pull request by evgenykochetkov**
Closes #830

### CODE REVIEW

File: packages/xod-client-electron/src/view/containers/App.jsx [hunk 1]
```diff
@@ -288,6 +310,37 @@ class App extends client.App {
...
+  }
+
+  confirmUnsavedChanges() {
+    if (!this.props.hasUnsavedChanges) return true;
+
+    const clickedButtonId = dialog.showMessageBox({
+      message: 'Current project has unsaved changes',
+      buttons: ['Save', 'Discard', 'Cancel'],
+      cancelId: 2,
+    });
```
Comment (nkrkv) [inline packages/xod-client-electron/src/view/containers/App.jsx]: Let’s be more clear and native?!

```diff
-      message: 'Current project has unsaved changes',
+      message: 'Save changes to the current project before closing?',
+      detail: 'If you don’t save the project, changes will be lost',
+      type: 'warning',
```

**Comment by nkrkv** [comment]
Other than the dialog adjustment proposal, everything looks excellent.

File: packages/xod-client-electron/src/view/containers/App.jsx (hunk 1 above)
Comment (evgenykochetkov) [reply packages/xod-client-electron/src/view/containers/App.jsx]: done

**Comment by nkrkv** [comment]
Merge!

**Pull request merged**